lib/src/utils.dart

```diff
@@ -17,23 +17,19 @@ extension NullableStringUtils on String? {
 
 extension ValueUtils on Value {
   dynamic toPrimitive() {
-    if (hasNullValue()) {
-      return nullValue;
-    }
     if (hasNumberValue()) {
       return numberValue;
-    }
-    if (hasStringValue()) {
+    } else if (hasStringValue()) {
```

Comment:
Great question, this was something I was unsure about as well and I'm trying to get clarity on in this PR review process.

 I am unsure if multiple of the Value() checks _should_ return true, it seems like they should not. But there is nothing in the code AFAIK enforcing a Value from not having multiple value types inside it. 

Even before shifting to `else if` it would have just returned whatever one it found first. The ordering was arbitrary from what I understand, it still is arbitrary other than moving `null` to be last as I'm proposing it be the default.

Reply:
ooh awesome, that definition is super helpful, thank you!